src/pages/filter/D3Plot.svelte

```diff
@@ -0,0 +1,295 @@
+          .enter()
+          .append("path")
+          .attr("class", function (gesture: RecordingRepresentation) {
+            return "line " + gesture.gestureClass;
+          }) // 2 class for each line: 'line' and the group name
+          .attr("d", path)
+          .style("fill", "none")
+          .style("stroke", function (gesture: RecordingRepresentation) {
+            return color(gesture.gestureClass);
+          })
+          .style("opacity", function (gesture: RecordingRepresentation) {
+            return opacity(gesture.gestureClass);
+          })
+          .style("stroke-width", function (gesture: RecordingRepresentation) {
+            return strokeWidth(gesture.gestureClass);
+          })
+        } else {
+          // Update live path
+          const newLivePathLine = () => path(data.pop() as RecordingRepresentation);
+          // Animate
+          plot.select(".live").transition().duration(50).attr("d", newLivePathLine);
+        }   
+        return;
+      } else {
+        plotDrawn = true;
+      }
+      // Extract the list of dimensions we want to keep in the plot.
+      // The path function take a row of the csv as input, and return x and y coordinates of the line to draw for this raw.
+      function path(gesture: RecordingRepresentation) {
+        return d3.line()(
+          dimensions.map(function (axis: axis) {
+            return [x(axis) as number, y[axis](gesture[axis])];
+          })
+        );
+      }
+
+      // Draw the lines
+      plot
+        .selectAll() // black magic
+        .data(data)
+        .enter()
+        .append("path")
+        .attr("class", function (gesture: RecordingRepresentation) {
+          return "line " + gesture.gestureClass;
+        }) // 2 class for each line: 'line' and the group name
+        .attr("d", path)
+        .style("fill", "none")
+        .style("stroke", function (gesture: RecordingRepresentation) {
+          return color(gesture.gestureClass);
+        })
+        .style("opacity", function (gesture: RecordingRepresentation) {
+          return opacity(gesture.gestureClass);
+        })
+        .style("stroke-width", function (gesture: RecordingRepresentation) {
+          return strokeWidth(gesture.gestureClass);
+        })
+        .on("mouseover", highlight)
+        .on("mouseleave", doNotHighlight);
+
+      // Draw the axis:
+      plot
+        .selectAll()
+        // For each dimension of the dataset I add a 'g' element:
+        .data(dimensions)
+        .enter()
+        .append("g")
+        .attr("class", "axis")
+        // I translate this element to its right position on the x axis
+        .attr("transform", function (axis: axis) {
+          return "translate(" + x(axis) + ")";
+        })
+        // And I build the axis with the call function
+        .each(function (axis: axis) {
+          d3.select(this).call(d3.axisLeft(d3.scaleLinear()).ticks(4).scale(y[axis]));
```

Comment:
Here we get a type error because the keyword `this` is implicitly of type `any`. We can solve this issue by explicitly typing it 

```ts
      .each(function (this: any, axis: axis) {
        d3.select(this).call(d3.axisLeft(d3.scaleLinear()).ticks(4).scale(y[axis]));
      })
```
Or since we know that it's a SVGGraphicsElement we can be even more explicit
```ts
      .each(function (this: SVGGraphicsElement, axis: axis) {
        d3.select(this).call(d3.axisLeft(d3.scaleLinear()).ticks(4).scale(y[axis]));
      })
```